### PR TITLE
feat: new "at_policy" package for building policy services on atProtocol

### PR DESCRIPTION
**Pull request by gkc**
**- What I did**
- new package at_policy
- added at_policy to the build / test matrix in .github workflows
- added core unit tests

The at_policy library provides generic scaffolding for
- building policy management services
- with which policy enforcement endpoints communicate via atProtocol 
- using the atClient libraries
- and therefore get all the benefits of using atProtocol and the atClient libraries - outbound communication only, end-to-end encryption, using atSigns rather than IP addresses, etc

**- How I did it**
- Added core code
- Added tests
- Added main README
- Added examples and a README in the examples directory

**- How to verify it**
Tests pass

### CODE REVIEW

**Review by cpswan — DISMISSED** [review]
What's the plan to add this to the automated tests we run from [`at_libraries/.github/workflows/at_libraries.yaml`](https://github.com/atsign-foundation/at_libraries/blob/trunk/.github/workflows/at_libraries.yaml)?

**Comment by gkc** [comment]
> What's the plan to add this to the automated tests we run from [`at_libraries/.github/workflows/at_libraries.yaml`](https://github.com/atsign-foundation/at_libraries/blob/trunk/.github/workflows/at_libraries.yaml)?

[Already done](https://github.com/atsign-foundation/at_libraries/pull/731/files#diff-7f2762ee0a2041b4e4ebd823359896fde1dbf7b1a6e13d8799ac81146e8ddb5d)

**Comment by gkc** [comment]
@XavierChanth just noticed that I hadn't updated `example/bin/service.dart` to use PolicyService, which I added after the initial implementation in order to simplify the policy service development experience.

**Comment by gkc** [comment]
@XavierChanth just noticed that I hadn't updated `example/bin/service.dart` to use PolicyService, which I added after the initial implementation in order to simplify the policy service development experience ... I'll update this tomorrow morning, maybe this evening although unlikely

**Comment by gkc** [comment]
> @XavierChanth just noticed that I hadn't updated `example/bin/service.dart` to use PolicyService, which I added after the initial implementation in order to simplify the policy service development experience ... I'll update this tomorrow morning, maybe this evening although unlikely

Actually that's incorrect - `policy.dart` is using PolicyService - `service.dart` is not because it is the service which is calling the policy service. In a version 1.1 I may add a PolicyClient to make it easier to write clients of the policy service

File: packages/at_policy/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+## Usage
+See full worked examples in the [example](example/README.md) directory.
+
+## Open source usage and contributions
+
+This is freely licensed open source code, so feel free to use it as is, suggest
+changes or enhancements or create your own version.
+See [CONTRIBUTING.md](../../CONTRIBUTING.md) for
+detailed guidance on how to set up tools, tests and make a pull request.on.
```
Comment (xavierchanth) [inline packages/at_policy/README.md]: nit: typo

File: packages/at_policy/lib/src/policy/models.dart [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+  Map<String, dynamic> toJson() => _$PolicyResponseToJson(this);
+
+  static PolicyResponse fromJson(Map<String, dynamic> json) =>
+      _$PolicyResponseFromJson(json);
+
+  @override
+  String toString() => jsonPrettyPrinter.convert(toJson());
+}
+
+abstract class CoreDeviceInfo {
```
Comment (xavierchanth) [inline packages/at_policy/lib/src/policy/models.dart]: Maybe fromJson and toJson should be included as abstract members here?

File: packages/at_policy/pubspec.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  at_client: ^3.3.0
+  at_cli_commons: ^1.2.0
+  at_utils: ^3.0.19
+  json_annotation: ^4.9.0
+
+dependency_overrides:
+  dartssh2:
+    git:
+      url: https://github.com/atsign-foundation/dartssh2
+      ref: trunk
```
Comment (xavierchanth) [inline packages/at_policy/pubspec.yaml]: This can be removed

**Review by xavierchanth — DISMISSED** [review]

File: packages/at_policy/lib/src/policy/models.dart (hunk 2 above)
Comment (gkc) [reply packages/at_policy/lib/src/policy/models.dart]: Maybe in a future PR

File: packages/at_policy/README.md (hunk 1 above)
Comment (gkc) [reply packages/at_policy/README.md]: fixed

File: packages/at_policy/pubspec.yaml (hunk 3 above)
Comment (gkc) [reply packages/at_policy/pubspec.yaml]: done

**Review by xavierchanth — APPROVED** [review]

**Pull request merged**